src/Extension/CoreExtension.php

```diff
@@ -1697,7 +1703,12 @@ public static function getAttribute(Environment $env, Source $source, $object, $
 
         // object property
         if (Template::METHOD_CALL !== $type) {
-            if (isset($object->$item) || \array_key_exists((string) $item, (array) $object)) {
+            $propertyReflector = false;
+            if (match (true) {
+                isset($object->$item) => true,
+                ($propertyReflector = self::$propertyReflectors[$object::class][$item] ??= self::getPropertyReflector($object, $item))->isInitialized($object) => true,
```

Comment:
this looks wrong to me. The custom reflector used as fallback using arrayCast assumes that `getValue()` is called on the same `$object` due to the internal memoization

Reply:
getValue() should always return `null` because of the isset check just before
the only exception I know about is DateTime/DateTimeImmutable
the memoization is only for them
which means: what's wrong?

Reply:
is there a guarantee that the Reflector will always be calling `getValue()` on the object passed to `isInitialized` just before to have the right memoized value ? The way the code is written currently avoids doing a second array cast in the special reflector, but it makes it easier to introduce bugs by breaking this implicit invariant assumed by the implementation.

Reply:
We don't really care actually, because the memoized value is going to be always just `null`, except for DateTime(Immutable), and then this is just strings, so no memleak. This saves an array cast, which is quite cosly operation for getting a `null` value most of the time... All this is local so I see no maintenance issue either

Reply:
it is not about memory leaks. It is about using a value from an array cast of a **different** object, in case we break the implicit invariant of always calling `isInitialized` just before `getValue`